workbench-core/environments/src/services/environmentTypeService.ts

```diff
@@ -49,20 +53,24 @@ export default class EnvironmentTypeService {
 
   /**
    * List environment type objects from DDB
-   * @param pageSize - the number of environment type objects to get (optional)
-   * @param paginationToken - the token from the previous page for continuation (optional)
+   * @param request - pagination, filter and sorting parameters
    *
    * @returns environment type objects
    */
   public async listEnvironmentTypes(
-    pageSize?: number,
-    paginationToken?: string
-  ): Promise<{ data: EnvironmentType[]; paginationToken: string | undefined }> {
+    request: ListEnvironmentTypeRequest
+  ): Promise<{ data: EnvironmentType[]; paginationToken?: string }> {
+    const { filter, sort, pageSize, paginationToken } = request;
+    validateSingleSortAndFilter(filter, sort);
     let queryParams: QueryParams = {
       key: { name: 'resourceType', value: this._resourceType },
       index: 'getResourceByCreatedAt',
       limit: pageSize && pageSize >= 0 ? pageSize : DEFAULT_API_PAGE_SIZE
     };
+    const gsiNames = ['getResourceByName', 'getResourceByStatus'];
+    const filterQuery = getFilterQueryParams(filter, gsiNames);
+    const sortQuery = getSortQueryParams(sort, gsiNames);
+    queryParams = { ...queryParams, ...filterQuery, ...sortQuery };
```

Comment:
I assume `filterQuery` and `sortQuery` will never both be defined? Can we add a check for that?

Reply:
Also this is so neat I love it.

Reply:
the validation is applied before this, so at this point it should have a valid one, I also added a check to allow same field for sort and filter, I even added at some point multiple filter to one field but realized that our dynamo layer doesn't allow that